lib/src/Base/Geom/Mesh.cxx

```diff
@@ -223,16 +223,16 @@ Bool Mesh::checkPointInSimplexWithCoordinates(const Point & point,
     const Scalar x02 = vertices_(simplices_(index, 2), 0) - x0;
     const Scalar y02 = vertices_(simplices_(index, 2), 1) - y0;
     const Scalar det = (x02 * y01 - y02 * x01);
-    const Scalar x = point[0] - x0;
-    const Scalar y = point[1] - y0;
     if (det == 0.0)
     {
       return false;
     }
+    const Scalar x = point[0] - x0;
+    const Scalar y = point[1] - y0;
     coordinates = Point(3);
     coordinates[1] = (x02 * y - y02 * x) / det;
     coordinates[2] = (x * y01 - y * x01) / det;
-    coordinates[0] = 1.0 - coordinates[1] - coordinates[2];
+    coordinates[0] = 0.5 + (0.5 - coordinates[1] - coordinates[2]);
     return coordinates[0] >= 0.0 && coordinates[0] <= 1.0 && coordinates[1] >= 0.0 && coordinates[1] <= 1.0 && coordinates[2] >= 0.0 && coordinates[2] <= 1.0;
```

Comment:
With an epsilon, the test would become:
```
return coordinates[0] >= -epsilon && coordinates[0] <= 1.0+epsilon && coordinates[1] >= -epsilon && coordinates[1] <= 1.0+epsilon && coordinates[2] >= -epsilon && coordinates[2] <= 1.0+epsilon;
```
and here, as the coordinates are within [0,1], a default choice of epsilon=1e-14 should be ok. In fact, it depends on the condition number of the linear function mapping the global coordinates into the barycentric coordinates.